src/witan/models/dem/ccm/components_functions.clj

```diff
@@ -23,6 +25,33 @@
         (ds/rename-columns {col-to-avg avg-name})
         (wds/rollup :mean avg-name [:gss-code :sex :age]))))
 
+(defn jumpoffyr-method-trend
+  "Takes in a dataset with historical data, a column name for a trend to be 
+  plotted for, a name for the trend col and years to start and end plotting 
+  the trend for. Returns a dataset with a new col for the calculated trend"
+  [historical-data trend-col trend-out start-yr-avg end-yr-avg]
+  (let [hist-earliest-yr (reduce min (ds/column historical-data :year))
+        hist-latest-yr (reduce max (ds/column historical-data :year))
+        start-yr (s/validate (s/pred #(>= % hist-earliest-yr)) start-yr-avg)
+        end-yr (s/validate (s/pred #(<= % hist-latest-yr)) end-yr-avg)
+        grouped-data (->> (i/query-dataset historical-data
```

Comment:
In `grouped-data` the thread is actually not need bc it's threading only once. I think that's the kind of things that kibit would signal.
You can replace by:
```
(i/$group-by [:sex :gss-code :age]
                                  (i/query-dataset historical-data
                                                   {:year {:$gte start-yr
                                                           :$lte end-yr}}))
```